src/packet_processor.rs

```diff
@@ -0,0 +1,43 @@
+use super::game_state::player::{PlayerStateOperations};
+use super::packet_router;
+use super::messenger::{MessengerOperations};
+use super::packet::read;
+use std::collections::HashMap;
+use std::io::{Cursor};
+use std::sync::mpsc::{Receiver, Sender};
+use uuid::Uuid;
+
+pub enum PacketProcessorOperations {
+    Inbound(InboundPacketMessage)
+}
+
+#[derive(Debug)]
+pub struct InboundPacketMessage {
+    pub conn_id: Uuid,
+    pub cursor: Cursor<Vec<u8>>
+}
+
+pub fn start_inbound(
+    receiver: Receiver<PacketProcessorOperations>,
+    messenger: Sender<MessengerOperations>,
+    player_state: Sender<PlayerStateOperations>
```

Comment:
So ideally we wouldn't have to pass these things through here- and actually this is a good example of the benefits we get via this design pattern. The fact that we have to pass these through to here indicates that we don't have a good separation of concerns- the packet router was being given way too much information.

Our services should get references just to the services that it should know about at the start of the program. This dependency injection will be a huge help once we start writing tests, and also for being very explicit about the dependencies of each service

Reply:
This will be fixed once we turn more things down the line into services- the boundaries of services should be defined by what state they need to function. The smaller the dependencies on the state, the smaller the services, the simpler the services.